standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java

```diff
@@ -1100,10 +1100,17 @@ public Partition add_partition(Partition new_part) throws TException {
 
   public Partition add_partition(Partition new_part, EnvironmentContext envContext)
       throws TException {
-    if (new_part != null && !new_part.isSetCatName()) {
+    if (new_part == null || new_part.getDbName() == null) {
+      throw new MetaException("Partition/DB name cannot be null.");
+    }
+    if (!new_part.isSetCatName()) {
       new_part.setCatName(getDefaultCatalog(conf));
     }
-    Partition p = client.add_partition_with_environment_context(new_part, envContext);
+    AddPartitionsRequest addPartitionsReq = new AddPartitionsRequest(new_part.getDbName(), new_part.getTableName(),
+            new ArrayList<>(Arrays.asList(new_part)), false);
+    addPartitionsReq.setCatName(new_part.getCatName());
+    addPartitionsReq.setEnvironmentContext(envContext);
+    Partition p = client.add_partition_req(addPartitionsReq);
```

Comment:
Can we just call `client.add_partitions_req` instead of introducing a new similar method?

Reply:
Ack